worldedit-core/src/main/java/com/fastasyncworldedit/core/extent/filter/LinkedFilter.java

```diff
@@ -1,69 +1,102 @@
 package com.fastasyncworldedit.core.extent.filter;
 
-import com.fastasyncworldedit.core.extent.filter.block.DelegateFilter;
 import com.fastasyncworldedit.core.extent.filter.block.FilterBlock;
 import com.fastasyncworldedit.core.internal.simd.VectorizedFilter;
 import com.fastasyncworldedit.core.queue.Filter;
+import com.fastasyncworldedit.core.queue.IChunk;
+import com.sk89q.worldedit.regions.Region;
 import jdk.incubator.vector.ShortVector;
+import org.jetbrains.annotations.Nullable;
 
 /**
- * Filter which links two Filters together for single-filter-input operations.
+ * Filter which links two Filters together for single-filter-input operations. Left filter is operated first.
  *
- * @param <T> Parent which extends Filter
- * @param <S> Child which extends Filter
+ * @param <L> Left filter
+ * @param <R> Right filter
  */
-public sealed class LinkedFilter<T extends Filter, S extends Filter> extends DelegateFilter<T> {
+public sealed class LinkedFilter<L extends Filter, R extends Filter> implements Filter {
 
-    private final S child;
+    private final L left;
+    private final R right;
+
+    public LinkedFilter(L left, R right) {
+        this.left = left;
+        this.right = right;
+    }
 
     @SuppressWarnings({"unchecked", "rawtypes"}) // we defeated the type system
-    public static <T extends Filter, S extends Filter> LinkedFilter<? extends T, ? extends S> of(T parent, S child) {
-        if (parent instanceof VectorizedFilter p && child instanceof VectorizedFilter c) {
-            return new VectorizedLinkedFilter(p, c);
+    public static <L extends Filter, R extends Filter> LinkedFilter<? extends L, ? extends R> of(L left, R right) {
+        if (left instanceof VectorizedFilter l && right instanceof VectorizedFilter r) {
+            return new VectorizedLinkedFilter(l, r);
         }
-        return new LinkedFilter<>(parent, child);
+        return new LinkedFilter<>(left, right);
+    }
+
+    public L getLeft() {
+        return this.left;
+    }
+
+    public R getRight() {
+        return this.right;
+    }
+
+    @Override
+    public boolean appliesChunk(int chunkX, int chunkZ) {
+        return getLeft().appliesChunk(chunkX, chunkZ) && getRight().appliesChunk(chunkX, chunkZ);
     }
 
-    public LinkedFilter(T parent, S child) {
-        super(parent);
-        this.child = child;
+    @Override
+    public <T extends IChunk> T applyChunk(T chunk, @Nullable Region region) {
+        chunk = getLeft().applyChunk(chunk, region);
```

Comment:
`chunk` could be null after the call according to the docs, would it make sense to pass the original value to the right in that case?